litmus-portal/frontend/src/pages/ApplicationDashboard/index.tsx

```diff
@@ -156,219 +280,601 @@ const DashboardPage: React.FC = () => {
+                                },
+                              });
+                            } else {
+                              setReFetch(true);
+                              setCentralAllowGraphUpdate(true);
+                              setCentralBrushPosition(undefined);
+                            }
+                          } else {
+                            setSelectedDashboardInformation({
+                              ...selectedDashboardInformation,
+                              range: {
+                                startDate: INVALID_DATE,
+                                endDate: INVALID_DATE,
+                              },
+                              refreshInterval: DEFAULT_REFRESH_RATE,
+                              relativeTime: DEFAULT_RELATIVE_TIME_RANGE,
+                              promQueries: generatePromQueries(
+                                {
+                                  startDate: INVALID_DATE,
+                                  endDate: INVALID_DATE,
+                                },
+                                selectedDashboardInformation.metaData
+                                  ?.panel_groups ?? [],
+                                selectedDashboardInformation.chaosEventQueryTemplate,
+                                selectedDashboardInformation.chaosVerdictQueryTemplate
+                              ),
+                            });
+                            setSelectedEvents([]);
+                            setReFetch(true);
+                            setCentralAllowGraphUpdate(true);
+                            setCentralBrushPosition(undefined);
+                          }
+                          if (showPromQueryResponseLoader) {
+                            setShowPromQueryResponseLoader(false);
+                          }
+                        }}
+                        panel_group_id={panelGroup.panel_group_id}
+                        panel_group_name={panelGroup.panel_group_name}
+                        panels={panelGroup.panels}
+                        selectedPanels={selectedPanels}
+                        metricDataForGroup={
+                          promData.panelGroupQueryMap[index]
+                            ? promData.panelGroupQueryMap[index]
+                                .metricDataForGroup
+                            : []
+                        }
+                        chaosData={promData.chaosEventData.chaosData}
+                      />
+                    </div>
+                  )
+                )}
+            </div>
           </div>
-        </div>
+        )}
       </div>
-      {dataSourceStatus !== 'ACTIVE' ? (
+      {dataSourceStatus !== ACTIVE && (
         <DataSourceInactiveModal
           dataSourceStatus={dataSourceStatus}
           dashboardID={selectedDashboardInformation.id}
         />
-      ) : (
-        <div />
       )}
     </Scaffold>
```

Comment:
Please try to break this file into components and this index.tsx file has become quite big.If its possible then it will be great

Reply:
yes i have already broken all of it into chunks now whatever remains are independant blocks will check if something can be done for the panel groups

Reply:
Adding this as a TODO @whitetiger1399 because it will require a good amount of testing for checking the render cycle